shared/people/invite-friends/invite-contacts.native.tsx

```diff
@@ -0,0 +1,174 @@
+import * as React from 'react'
+import * as Kb from '../../common-adapters/mobile.native'
+import {Section as _Section} from '../../common-adapters/section-list'
+import useContacts from '../../teams/invite-by-contact/use-contacts.native'
+import {ContactProps as Contact} from '../../teams/invite-by-contact/index.native'
+import {memoize} from '../../util/memoize'
+import * as Container from '../../util/container'
+import * as Styles from '../../styles'
+import {mapGetEnsureValue} from '../../util/map'
+
+type Section = _Section<Contact, {title: string}>
+
+const categorize = (contact: Contact): string => {
+  if (!contact.name) {
+    return '0-9'
+  }
+  const firstLetter = contact.name[0].toUpperCase()
+  if ('ABCDEFGHIJKLMNOPQRSTUVWXYZ'.includes(firstLetter)) {
+    return firstLetter
+  } else if ('0123456789'.includes(firstLetter)) {
+    return '0-9'
+  } else {
+    return 'Other'
+  }
+}
+const filterAndSectionContacts = memoize((contacts: Contact[], search: string): Section[] => {
+  const sectionMap: Map<string, Contact[]> = new Map()
+  contacts
+    .filter(contact => contact.name.toLowerCase().includes(search) || contact.value.includes(search))
+    .forEach(contact => {
+      const category = categorize(contact)
+      const section = mapGetEnsureValue(sectionMap, category, [])
+      section.push(contact)
+    })
+  const sections: Section[] = []
+  for (const letter of 'ABCDEFGHIJKLMNOPQRSTUVWXYZ') {
+    if (sectionMap.has(letter)) {
+      sections.push({
+        data: sectionMap.get(letter)!,
+        key: letter,
+        title: letter,
+      })
+    }
+  }
+  for (const sectionKey of ['0-9', 'Other']) {
+    if (sectionMap.has(sectionKey)) {
+      sections.push({
+        data: sectionMap.get(sectionKey)!,
+        key: sectionKey,
+        title: sectionKey,
+      })
+    }
+  }
+  return sections
+})
+
+const InviteContacts = () => {
+  const {contacts, errorMessage, loading} = useContacts()
+  const [search, setSearch] = React.useState('')
+  const [selectedContacts, setSelectedContacts] = React.useState(new Set<string>())
```

Comment:
want to differentiate emails and phone numbers in here?

Reply:
I'd rather differentiate on submit.

Reply:
heh actually just tried doing the submit code for this and you're totally right. WIll do it in the other PR though unless you have more changes for me here.

Reply:
oh, we raced. Will do here with other changes.